pkg/oauth2client/oauth_client.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+//go:generate mockgen -destination oauth_client_mocks_test.go -self_package mocks -package oauth2client_test -source=oauth_client.go -mock_names OAuth2Client=MockOAuth2Client
+
+package oauth2client
+
+import (
+	"context"
+	"net/http"
+
+	"golang.org/x/oauth2"
+)
+
+type Client struct {
+}
+
+func NewOAuth2Client() *Client {
+	return &Client{}
+}
+
+func (c *Client) Exchange(
+	ctx context.Context,
+	cfg oauth2.Config,
+	code string,
+	opts ...oauth2.AuthCodeOption,
+) (*oauth2.Token, error) {
+	return (&cfg).Exchange(ctx, code, opts...)
+}
+
+func (c *Client) ExchangeWithCustomClient(
```

Comment:
consider moving `client *http.Client` to the method above and just have a single `Exchange` func that accepts `client *http.Client` - for now there is no need for that versatility

Reply:
Done @aholovko